ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/perception/RDXFoundationPoseDemo.java

```diff
@@ -0,0 +1,297 @@
+package us.ihmc.rdx.perception;
+
+import imgui.ImGui;
+import org.bytedeco.javacpp.BytePointer;
+import org.bytedeco.opencv.global.opencv_imgcodecs;
+import org.bytedeco.opencv.opencv_core.Mat;
+import perception_msgs.msg.dds.FoundationPoseRequest;
+import perception_msgs.msg.dds.FoundationPoseResult;
+import perception_msgs.msg.dds.ImageMessage;
+import us.ihmc.commons.thread.RepeatingTaskThread;
+import us.ihmc.communication.PerceptionAPI;
+import us.ihmc.communication.ros2.sync.ROS2PeerClockOffsetEstimator;
+import us.ihmc.perception.RawImage;
+import us.ihmc.perception.cuda.CUDATools;
+import us.ihmc.perception.detections.InstantDetection;
+import us.ihmc.perception.detections.yolo.YOLOv8DetectionThread;
+import us.ihmc.perception.detections.yolo.YOLOv8InstantDetection;
+import us.ihmc.perception.imageMessage.CompressionType;
+import us.ihmc.perception.imageMessage.PixelFormat;
+import us.ihmc.perception.tools.PerceptionMessageTools;
+import us.ihmc.rdx.Lwjgl3ApplicationAdapter;
+import us.ihmc.rdx.ui.RDXBaseUI;
+import us.ihmc.rdx.ui.gizmo.RDXPose3DGizmo;
+import us.ihmc.rdx.ui.graphics.RDXRawImagePointCloudVisualizer;
+import us.ihmc.rdx.ui.graphics.RDXReferenceFrameGraphic;
+import us.ihmc.rdx.ui.graphics.ros2.yolo.RDXROS2YOLOv8Visualizer;
+import us.ihmc.ros2.ROS2Node;
+import us.ihmc.ros2.ROS2NodeBuilder;
+import us.ihmc.ros2.ROS2Publisher;
+import us.ihmc.ros2.ROS2QosProfile;
+import us.ihmc.ros2.ROS2Topic;
+import us.ihmc.sensors.ImageSensor;
+import us.ihmc.sensors.zed.ZEDImageSensor;
+import us.ihmc.sensors.zed.ZEDModelData;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static us.ihmc.zed.global.zed.*;
+
+class RDXFoundationPoseDemo
```

Comment:
Should be public. Also can you add some Javadoc to just the class to explain how this fits in, why it exists?